rustls/src/msgs/hsjoiner.rs

```diff
@@ -75,12 +75,15 @@ impl HandshakeJoiner {
     /// enough to contain a header, and that header has a length which falls
     /// within `buf`.
     fn buf_contains_message(&self) -> bool {
-        self.buf.len() >= HEADER_SIZE
-            && self.buf.len()
-                >= (codec::u24::decode(&self.buf[1..4])
-                    .unwrap()
-                    .0 as usize)
-                    + HEADER_SIZE
+        if self.buf.len() < HEADER_SIZE {
+            return false;
+        }
+
+        let (header, rest) = self.buf.split_at(HEADER_SIZE);
+        match codec::u24::decode(&header[1..]) {
```

Comment:
I agree the correctness of this version is clearer than the original version; i.e. this is an improvement. The PR that tried to get rid of the `unwrap()` in the wrong way is evidence of this. So, I think we should make this change.

I think we should strive to later improve upon this by getting rid of the use of panicking primitives like `split_at` and `[]` but that's not urgent to do now.